src/aggregator/aggregator/follower_flush_mgr_test.go

```diff
@@ -131,6 +132,63 @@ func TestFollowerFlushManagerCanNotLeadForwardedFlushWindowsNotEnded(t *testing.
 	require.False(t, mgr.CanLead())
 }
 
+func TestFollowerFlushManagerCanLeadNotFlushed(t *testing.T) {
+	window := 10 * time.Minute
+	testFlushTimes := &schema.ShardSetFlushTimes{
+		ByShard: map[uint32]*schema.ShardFlushTimes{
+			123: &schema.ShardFlushTimes{
+				StandardByResolution: map[int64]int64{
+					window.Nanoseconds(): 0,
+				},
+				ForwardedByResolution: map[int64]*schema.ForwardedFlushTimesForResolution{
+					window.Nanoseconds(): &schema.ForwardedFlushTimesForResolution{
+						ByNumForwardedTimes: map[int32]int64{
+							1: 0,
+						},
+					},
+				},
+			},
+		},
+	}
+
+	runTestFn := func(
+		t *testing.T,
+		followerOpenedAt time.Time,
+		expectedCanLead bool,
+	) {
+		ctrl := gomock.NewController(t)
+		defer ctrl.Finish()
+
+		doneCh := make(chan struct{})
+		electionManager := NewMockElectionManager(ctrl)
+		electionManager.EXPECT().IsCampaigning().Return(true).AnyTimes()
+		opts := NewFlushManagerOptions().SetElectionManager(electionManager)
+		mgr := newFollowerFlushManager(doneCh, opts).(*followerFlushManager)
+
+		mgr.processed = testFlushTimes
+		mgr.openedAt = followerOpenedAt
+		require.Equal(t, expectedCanLead, mgr.CanLead())
+	}
+
+	t.Run("opened_on_the_window_start", func(t *testing.T) {
+		followerOpenedAt := time.Now().Truncate(window)
```

Comment:
These tests don't flake do they? Not sure if worth it, but we could use a mocked out time for this to make it super clear what times are when (`time.Now()` is a bit more opaque). Either way though.

Reply:
They are not flaky, but I agree, for clarity it's better to pass the mocked fixed now.